llama_hub/microsoft_sharepoint/base.py

```diff
@@ -28,6 +28,8 @@ def __init__(
         client_id: str,
         client_secret: str,
         tenant_id: str,
+        filename_as_id: bool = False,
+        file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = None
```

Comment:
I think there's a small mistake here:
`file_extractor: Optional[Dict[str, BaseReader]] = None,`

For ref: https://github.com/run-llama/llama_index/blob/0393b081f3aed854e0a628f49b8e51f8da7906ef/llama_index/readers/file/base.py#L118

Reply:
Hopefully my second commit should solve the issue that prevented the first run to work out... previously I just forgot to add the appropriate imports from typing (Optional and Union).

Reply:
You don't have to add `union` for basereader.  It'll be always a baseReader class 

Replace file_extractor line with the below one 

`file_extractor: Optional[Dict[str, BaseReader]] = None,`

Reply:
I wrote it that way because I was shamelessly copying line **25** of `llama_hub/minio/minio-client/base.py`

`        file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = None,`

however, I'll rewrite it that way